### PR TITLE
Avoid duplicate (and unnecessary) RPCs for reading each counter value, a...

### PR DESCRIPTION
**Pull request by anwar6953**
...s well as for reading the status.

Build: http://builds.cask.co/browse/CDAP-DUT1256-1 (GREEN)
JIRA: https://issues.cask.co/browse/CDAP-85

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java [hunk 1]
```diff
@@ -59,14 +60,16 @@ public MapReduceMetricsWriter(Job jobConf, BasicMapReduceContext context) {
...
-    reportSystemStats();
+    Counters jobCounters = jobConf.getCounters();
+    reportMapredStats(jobCounters);
+    reportSystemStats(jobCounters);
   }
 
   // job level stats from counters built in to mapreduce
-  private void reportMapredStats() throws IOException, InterruptedException {
+  private void reportMapredStats(Counters jobCounters) throws IOException, InterruptedException {
+    JobStatus jobStatus = jobConf.getStatus();
```
Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java]: does it make sense to also fetch jobStatus once?

**Comment by abaranau** [comment]
one Q, otherwise LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java (hunk 1 above)
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java]: Yes, that's what I also did in this PR.
I feel that each time we call `reportStats`, it should just report the stats as if it were a snapshot (as opposed to calling `job.getStatus` multiple times throughout the reporting iteration.

**Comment by abaranau** [comment]
merge

**Pull request merged**